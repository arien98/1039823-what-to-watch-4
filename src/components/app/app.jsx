import React from "react";
import PropTypes from "prop-types";
import Main from "../main/main.jsx";

const App = (props) => {
  const {descFilm, filmsTitles, filmsData} = props;

  return <Main
    descFilm={descFilm}
    filmsTitles={filmsTitles}
    filmsData={filmsData}
  />;
};

App.propTypes = {
  descFilm: PropTypes.object,
  filmsTitles: PropTypes.arrayOf(PropTypes.string),
  filmsData: PropTypes.arrayOf(PropTypes.object),
};

export default App;
