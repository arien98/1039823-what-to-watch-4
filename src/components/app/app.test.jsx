import React from "react";
import renderer from "react-test-renderer";
import {App} from "./app";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";

const mockStore = configureStore([]);

describe(`app renders correctly`, () => {
  test(`it renders correctly`, () => {
    const props = {
      onFilmCardClick: () => {},
      screenMode: `main page`,
      promoFilm: {
        title: `Aviator`,
        poster: `../img/aviator.jpg`,
        genre: `Drama`,
        releaseDate: 2010,
      },
      filmsData: [{
        id: 2,
        title: `Bohemian Rhapsody`,
        poster: `img/bohemian-rhapsody.jpg`,
        genre: `Dramas`,
      },
      {
        id: 3,
        title: `Macbeth`,
        poster: `img/macbeth.jpg`,
        genre: `Dramas`,
      },
      {
        id: 4,
        title: `Aviator`,
        poster: `img/aviator.jpg`,
        genre: `Horror`,
      },
      ],
    };

    const store = mockStore({
      // screenMode: `main page`,
      // promoFilm: {
      //   title: `Aviator`,
      //   poster: `../img/aviator.jpg`,
      //   genre: `Drama`,
      //   releaseDate: 2010,
      // },

      // filmsData: [{
      //   id: 2,
      //   title: `Bohemian Rhapsody`,
      //   poster: `img/bohemian-rhapsody.jpg`,
      //   genre: `Dramas`,
      // },
      // {
      //   id: 3,
      //   title: `Macbeth`,
      //   poster: `img/macbeth.jpg`,
      //   genre: `Dramas`,
      // },
      // {
      //   id: 4,
      //   title: `Aviator`,
      //   poster: `img/aviator.jpg`,
      //   genre: `Horror`,
      // },
      // ],
    });

    const tree = renderer
      .create(
          <Provider store={store}>
            <App {...props}/>
          </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
