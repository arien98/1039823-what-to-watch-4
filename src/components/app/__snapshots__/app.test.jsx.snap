// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app renders correctly it renders correctly 1`] = `null`;
